File: simulator/effects/combatant_effect.py
from ..effects.effect import Effect


class CombatantEffect(Effect):

    def __init__(self, initiator, combatants):
        Effect.__init__(self, initiator)
        self.combatants = combatants

    def is_affecting(self, combatant):
        return combatant in self.combatants

File: c++/effects/combatant_effect.cpp
#include "effects/combatant_effect.hpp"
#include <algorithm>

namespace enc
{
  CombatantEffect::CombatantEffect(Combatant* initiator, const std::vector<Combatant*>& combatants)
    : Effect(initiator),
      _combatants(combatants)
  {
  }

  bool CombatantEffect::isAffecting(Combatant* combatant) const
  {
    return std::find(_combatants.begin(), _combatants.end(), combatant) != _combatants.end();
  }

} // namespace enc
